packages/transit-vehicle-overlay/src/components/markers/RotatedVehicleMarker/index.js

```diff
@@ -0,0 +1,58 @@
+import React from "react";
+import PropTypes from "prop-types";
+
+import { transitVehicleType } from "@opentripplanner/core-utils/lib/types";
+import RotatedMarker from "../RotatedMarker";
+import * as utils from "../../../utils";
+
+/**
+ * renders circular leaflet markers for the vehicles
+ * at higher zoom levels, the icon used in the marker is based on the vehicles mode information
+ */
+export default function RotatedVehicleMarker({
+  children,
+  icon,
+  isTracked,
+  onVehicleClicked,
+  vehicle
+}) {
+  const { lat, lon, heading } = vehicle;
+
+  return (
+    <RotatedMarker
+      icon={icon}
+      position={[lat, lon]}
+      rotationAngle={heading}
+      rotationOrigin="center center"
+      onClick={() => onVehicleClicked(vehicle, isTracked)}
```

Comment:
I would really prefer that we not define anonymous functions like this for props. Especially if we're talking about a component that could get rendered hundreds of time on a map.